CONTRIBUTING.md

```diff
@@ -1,297 +1,385 @@
+
+##### Work On All Locales
+
+Scripts must work on every locale available for Windows. Be careful when using text strings for windows in the script that could differ in another language.
+
+##### Avoid Brittle Scripts
+
+Do not create brittle scripts that work only when the user doesn't interfere. All script elements should be as precise as possible - for instance, instead of using the [Send](https://autohotkey.com/docs/commands/Send.htm) function, which will work correctly only if the desired window is active, use [ControlSend](https://autohotkey.com/docs/commands/ControlSend.htm) which doesn't require window activation or use [BlockInput](https://autohotkey.com/docs/commands/BlockInput.htm) for short periods.
+
+#### Source Files
+
+##### Encoding
 
 Always __use UTF-8 without BOM__ for the `*.nuspec` and __UTF-8 with BOM__ for the `*.ps1` files. See [character encodings](https://docs.chocolatey.org/en-us/create/create-packages#character-encoding).
 
-### 2.2 Code style
+##### Code Style
+
+Refrain from committing code with obvious styling problems such as irregular indentation levels, very long lines, too many comments, too many empty lines, and other styling issues. Please follow the [PoshCode PowerShell Practice and Style Guide](https://github.com/PoshCode/PowerShellPracticeAndStyle).
+
+The project contains a [`.editorconfig`](https://github.com/chocolatey-community/chocolatey-packages/blob/master/.editorconfig)
+ file that you can use with many editors via [EditorConfig](http://editorconfig.org/) plugins. Ensure you use an editor with this support and have the support enabled.
 
-Do not commit code with obvious styling problems such as irregular indentation levels, very long lines, too many comments, too many empty lines, etc.
+Keep the package source files clean and remove outdated code and unnecessary comments. Comment on non-obvious code so that others can easily understand what it does.
 
-The project contains [.editorconfig](https://github.com/chocolatey-community/chocolatey-packages/blob/master/.editorconfig)
- file that can be used with many editors via [EditorConfig](http://editorconfig.org/) plugins.
+### Update Script
 
-Keep the package source files clean and remove obsolete or outdated code and unnecessary comments. Comment non-obvious code so that others can easily understand what it does.
+Update scripts are the PowerShell scripts responsible for getting information about the software's most current version, where it can be acquired from, and how to modify the source code to produce the package. It is typically called `update.ps1`.
+These scripts should be located in the root of the package directory, right next to the metadata (`.nuspec`) file.
+The update script should be compatible with PowerShell v5 and may not need to be compatible with PowerShell Core.
```

Comment:
```suggestion
The update script should be compatible with Windows PowerShell v5 and may not need to be compatible with PowerShell Core.
```

I feel like it's cleaner to specify Windows PowerShell for anything before 6.0+, but I'm not sure the best way to refer to newer ones as 7.0 dropped the `Core` nomenclature, but most people still know it as Core (and generally it's understood what is meant).

Reply:
I think Core is still fine to reference it as. If you run `$PSVersionTable` it still says the edition being Core, if I recall correctly.